.circleci/config.yml

```diff
@@ -90,20 +93,14 @@ orbs:
     jobs:
       checkout:
         <<: *job_defaults
-        parameters:
-          ruby_version:
-            description: Ruby version
-            type: string
-          image:
-            description: Docker image location
-            type: string
         docker:
-          - <<: *container_base
-            image: <<parameters.image>>
+          - image: alpine/git:v2.26.2 # Minimal image with git available
+            user: root # We currently require root to write to `/app`
+        resource_class: small
         steps:
           - checkout
           - save_cache:
```

Comment:
You can also use a workspace here instead, might be easier than managing cache keys?

https://circleci.com/docs/2.0/workflows/#using-workspaces-to-share-data-among-jobs

Reply:
We use workspace for the CI coverage collection, but it's so much slower for the same amount data than using cache keys.

Unless we need to restore files in a different directory (which we need for CI coverage results), using the cache is much faster.

Also, cache keys allow for partial key matching (important for cache restoration in new branches), but this wouldn't apply to the checkout step.

At the end of the day, I removed the job altogether and moved it to just another step in the `build-2.*` jobs.

Reply:
👍🏻